src/protocols.c

```diff
@@ -170,19 +186,75 @@ static int read_file(const char *path) {
 
   return status;
 } /* int read_file */
+#endif
+
+#if KERNEL_SOLARIS && HAVE_KSTAT_H
+/*
+ * Retrieves all available key/value pairs for IP, ICMP, UDP and TCP from kstat
+ * modules ip, icmp, tcp and udp
+ */
+static int read_kstat(const char *mod_name) {
+  extern kstat_ctl_t *kc;
+  kstat_t *ksp_chain = NULL;
+
+  if (mod_name == NULL)
+    return -1;
+
+  if (kc == NULL)
+    return -1;
+  for (ksp_chain = kc->kc_chain; ksp_chain != NULL;
+       ksp_chain = ksp_chain->ks_next) {
+    if (strncmp(ksp_chain->ks_module, mod_name, sizeof(ksp_chain->ks_module)) ==
```

Comment:
Is `ks_module` terminated with a nul byte? If so, use *strcmp()* instead -- *strncmp()* implies (to me) that the code is trying to do a prefix match.